### PR TITLE
linux: fix unused option warnings on 5.x kernels

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Description of changes

Rebase of #164906 by @brandonweeks 

> Fixes:
>
> ```
> warning: unused option: DEBUG_STACKOVERFLOW
> warning: unused option: EXT4_ENCRYPTION
> warning: unused option: F2FS_FS_ENCRYPTION
> warning: unused option: RCU_PERF_TEST
> warning: unused option: XEN_SCRUB_PAGES
> warning: unused option: XEN_SELFBALLOONING
> warning: unused option: XEN_STUB
> warning: unused option: XEN_TMEM
> ```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabianhjr** [comment]
Tested with 6.0.9-hardened1

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 202460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.kernels.linux_4_14_hardened</li>
    <li>linuxKernel.kernels.linux_4_19_hardened</li>
    <li>linuxKernel.kernels.linux_libre</li>
    <li>linuxKernel.kernels.linux_rt_5_4</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>linuxKernel.kernels.linux_4_14</li>
    <li>linuxKernel.kernels.linux_4_19</li>
    <li>linuxKernel.kernels.linux_5_10</li>
    <li>linuxKernel.kernels.linux_5_10_hardened</li>
    <li>linuxKernel.kernels.linux_5_15</li>
    <li>linuxKernel.kernels.linux_hardened (linuxKernel.kernels.linux_5_15_hardened)</li>
    <li>linuxKernel.kernels.linux_5_4</li>
    <li>linuxKernel.kernels.linux_6_0</li>
    <li>linuxKernel.kernels.linux_6_0_hardened</li>
    <li>linuxKernel.kernels.linux_latest_libre</li>
    <li>linuxKernel.kernels.linux_lqx</li>
    <li>linuxKernel.kernels.linux_rt_5_10</li>
    <li>linuxKernel.kernels.linux_testing_bcachefs</li>
    <li>linuxKernel.kernels.linux_xanmod</li>
    <li>linuxKernel.kernels.linux_xanmod_latest (linuxKernel.kernels.linux_xanmod_stable)</li>
    <li>linuxKernel.kernels.linux_zen</li>
  </ul>
</details>

**Comment by lovesegfault** [comment]
Checking whether the failures happen on master

**Comment by lovesegfault** [comment]
`linuxKernel.kernels.linux_libre` and `linuxKernel.kernels.linux_rt_5_4` seem to build AOK, let me try again on this PR.

**Comment by lovesegfault** [comment]
Both failures are spurious, this doesn't break any builds
